tracer/tracer.go

```diff
@@ -0,0 +1,166 @@
+package tracer
+
+// TODO: change package
+
+import (
+	"time"
+
+	opentracing "github.com/opentracing/opentracing-go"
+)
+
+// Tracer extends the opentracing.Tracer interface with methods to
+// probe implementation state, for use by tracer consumers.
+type Tracer interface {
+	opentracing.Tracer
+	Options() Options
+}
+
+type SpanRecorder interface {
+	RecordSpan(span RawSpan)
+}
+
+type Options struct {
+	ShouldSample   func(traceID string) bool
+	Recorder       SpanRecorder
+	EnableSpanPool bool
+}
+
+// TODO: review ShouldSample
```

Comment:
We would want sampling implemented off the bat. Please look into how we do it in Java and adapt accordingly for Go: https://github.com/wavefrontHQ/wavefront-opentracing-sdk-java/blob/master/src/main/java/com/wavefront/opentracing/WavefrontTracer.java#L348

You can add the sampler to this SDK itself. We put it under `wavefront-sdk-java` for Java as we needed to reuse on the proxy.